app/controllers/downloads_controller.rb

```diff
@@ -2,6 +2,15 @@ class DownloadsController < ApplicationController
   def show
     #.available_repos
     repo = current_user.repos.find(params[:id])
-    send_data Download.from_s3(repo), :filename => repo.name
+    if repo.password.nil? || password_check
+      #instead of test.jpg we will use repo.name
+      send_data Download.from_s3(repo), :filename => "test.jpg"
```

Comment:
@ethanbennett I think this piece has to be modified as `:filename => "test.jpg"` will have our old bug back to downloading each file named `test.jpg`. It should be `:filename => repo.name`

FYI @DavidKnott